stepper2.md

```diff
@@ -0,0 +1,212 @@
+Stepper Motor 2.0
+===
+
+Provides support for full 2 wire, full 3 wire, full 4 wire, half 3 wire, and half 4 wire stepper motor drivers (H-bridge, darlington array, etc) as well as step + direction drivers such as the [EasyDriver](http://www.schmalzhaus.com/EasyDriver/).
+Current implementation supports 10 stepper motors at the same time (#[0-9]).
+
+Includes optional support for acceleration and deceleration of the motor.
+
+Also includes multiStepper support which allows groups of steppers to be simultaneously passed different ```to``` values and the duration of their movements will match. Up to five multiStepper groups can be created. The total number of steppers is still limited to 10.
+
+Example files:
+ * The Stepper feature is include by default in [ConfigurableFirmata.ino](https://github.com/firmata/ConfigurableFirmata/blob/master/examples/ConfigurableFirmata/ConfigurableFirmata.ino).
+ * [Example implementation](https://github.com/firmata/ConfigurableFirmata/blob/master/src/StepperFirmata.cpp) as a configurable Firmata feature class.
+
+Protocol
+---
+
+**Stepper configuration**
+
+*Note: `stepDelay` is the the number of microseconds between steps. The default
+value is 1us. You can change the delay to 2us (useful for high current stepper
+motor drivers). Additional delay values can be added in the future.*
+```
+0  START_SYSEX                                (0xF0)
+1  Stepper Command                            (0x62)
+2  config command                             (0x00 = config)
+3  device number                              (0-9) (Supports up to 10 motors)
+
+4  interface                                  (upper 3 bits = wire count:
+                                                000XXXX = driver
+                                                010XXXX = two wire
+                                                011XXXX = three wire
+                                                100XXXX = four wire)
+
+                                              (4th - 6th bits = step type
+                                                XXX000X = whole step
+                                                XXX001X = half step
+                                                XXX010X = micro step)
+
+                                              (lower 1 bit = has enable pin:
+                                                XXXXXX0 = no enable pin
+                                                XXXXXX1 = has enable pin)
+
+5  stepDelay                                  (0-127)
```

Comment:
Looking at AccelStepper.cpp minPulseWidth serves the exact same purpose that stepDelay did in my driver class so we can safely remove stepDelay here.